x-pack/legacy/plugins/maps/public/layers/util/es_agg_utils.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { i18n } from '@kbn/i18n';
+import _ from 'lodash';
+import { IndexPattern, IFieldType } from '../../../../../../../src/plugins/data/public';
+
+export function getField(indexPattern: IndexPattern, fieldName: string) {
+  const field = indexPattern.fields.getByName(fieldName);
+  if (!field) {
+    throw new Error(
+      i18n.translate('xpack.maps.source.esSearch.fieldNotFoundMsg', {
+        defaultMessage: `Unable to find '{fieldName}' in index-pattern '{indexPatternTitle}'.`,
+        values: { fieldName, indexPatternTitle: indexPattern.title },
+      })
+    );
+  }
+  return field;
+}
+
+export function addFieldToDSL(dsl: object, field: IFieldType) {
+  return !field.scripted
+    ? { ...dsl, field: field.name }
+    : {
+        ...dsl,
+        script: {
+          source: field.script,
+          lang: field.lang,
+        },
+      };
+}
+
+export function extractPropertiesFromBucket(bucket: any, ignoreKeys: string[] = []) {
+  const properties: Record<string | number, unknown> = {};
+  for (const key in bucket) {
```

Comment:
can you add a `bucket.hasOwnProperty` check (guideline wrt avoiding prototype pollution).